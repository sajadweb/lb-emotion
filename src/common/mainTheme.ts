import { Theme } from './styled';
import * as color from 'color';
function darken(hex: string) {
    return color(hex)
        .darken(0.03)
        .hex();
}
const mainTheme: Theme = {
    color: {
        primary: '#00aba9',
        primary_dark: darken('#00aba9'),
        secondary: '#24292e',
        secondary_dark: darken('#24292e'),
        warning: '#f0ad4e',
        warning_dark: darken('#f0ad4e'),
        light: '#f8f9fa',
        light_dark: darken('#f8f9fa'),
        dark: '#000',
        success: '#d4edda',
        success_dark: '#155724',
        info: '#5bc0de',
        info_dark: darken('#5bc0de'),
        danger: '#d9534f',
        danger_dark: darken('#d9534f'),
    },
    font: {
        size: 16,
        family: '"Roboto"',
    },
    darken: (_color?: string) => {
        return color(_color)
            .darken(0.03)
            .hex();
    },
    hexToRgba: (hex: string, a: number) => {
        return color(hex).fade(a ? a : 0.5).string();
    }
};

export default mainTheme;
