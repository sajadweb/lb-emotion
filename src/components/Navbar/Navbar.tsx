import styled from '../../common/styled';
import Variables from '../Variables';
/**
 * @description it is props for Navbar component
 */
interface NavbarProps extends Variables.Base {
    backgroundColor?: string;
    color?: string;
    noShadow?: boolean;
}

const Navbar = styled('div')(
    (props: (NavbarProps | any)) => {
        // const base = `
        // background-color: #f8f9fa!important;1
        // -ms-flex-flow: row nowrap;1
        // flex-flow: row nowrap;1
        // -ms-flex-pack: start;1
        // justify-content: flex-start;
        // position: fixed;
        // top: 0;
        // left: 0;
        // right:0;
        // min-height: 4rem;
        // display: -ms-flexbox;
        // display: flex;
        // -ms-flex-wrap: wrap;
        // flex-wrap: wrap;
        // -ms-flex-align: center;
        // align-items: center;
        // -ms-flex-pack: justify;
        // justify-content: space-between;
        // padding: .5rem 1rem;
        // *, ::after, ::before {
        //     box-sizing: border-box;
        // }
        // `;
        const base = {
            ...[`;
            position: -webkit-fixed;
            position: fixed;
            display: -ms-flexbox;
            display: flex;
            `],
            '-ms-flex-flow': 'row nowrap',
            'flex-flow': 'row nowrap',
            '-ms-flex-pack': 'start',
            justifyContent: 'flex-start',
            top: 0,
            left: 0,
            right: 0,
            // minHeight: '4rem',
            '-ms-flex-wrap': 'wrap',
            'flex-wrap': 'wrap',
            '-ms-flex-align': 'center',
            'flex-align': 'center',
            'justify-content': 'space-between',
            // width: '100%',
            padding: '1.3rem 1rem',
            // margin: 'auto',
            backgroundColor: 'transparent',
            color: '#000'
        };
        if (props.bgColor) {
            if (props.theme.color[props.bgColor]) {
                base.backgroundColor = props.theme.color[props.bgColor];
            } else {
                base.backgroundColor = props.bgColor;
            }
        }
        base.color = props.color ? props.color : base.color;
        return base;
    }
);

export default Navbar;
