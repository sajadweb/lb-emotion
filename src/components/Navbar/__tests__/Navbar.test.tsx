import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Navbar from '../Navbar';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const tree = renderer.create(<Navbar theme={mainTheme} primary>hi test</Navbar>).toJSON();
    // expect(tree).toHaveStyleRule('background-color', mainTheme.color.primary);
    // expect(tree).toHaveStyleRule('color', mainTheme.color.primary_dark);
    expect(tree).toMatchSnapshot();
});
