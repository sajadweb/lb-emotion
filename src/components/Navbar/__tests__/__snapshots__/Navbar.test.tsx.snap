// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<div
  className="css-m8k7v0"
>
  hi test
</div>
`;
